pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineScript.groovy

```diff
@@ -40,12 +41,38 @@ public class DockerPipelineScript extends DeclarativeAgentScript<DockerPipeline>
 
     @Override
     public Closure run(Closure body) {
-        String targetLabel = describable.label
-        if (targetLabel == null) {
-            targetLabel = script.dockerLabel()?.trim()
+        if (describable.reuseNode && script.getContext(FilePath.class) != null) {
```

Comment:
These changes looks like a copy of the changes in `DockerPipelineFromDockerfileScript` can't we get some sharing going on like you did with the config?
The only difference seems to be in `runImage` so an `AbstractDeclarativeAgentScript` with all the common bits and an `abstract runImage` would be good.